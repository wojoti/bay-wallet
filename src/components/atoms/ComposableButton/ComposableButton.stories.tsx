import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {ComposableButton} from 'components/atoms';
import {Text} from 'react-native';
import {primary} from 'theme/colors';

export default {
  title: 'Atoms/ComposableButton',
  component: ComposableButton,
  argTypes: {
    onPress: {action: 'onPress'},
  },
} as ComponentMeta<typeof ComposableButton>;
const Template: ComponentStory<typeof ComposableButton> = args => (
  <ComposableButton {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  testId: '',
  children: <Text>Added required component</Text>,
};

export const Secondary = Template.bind({});
Secondary.args = {
  testId: '',
  customStyle: {backgroundColor: primary, borderWidth: 0},
  children: <Text>Added required component</Text>,
};
