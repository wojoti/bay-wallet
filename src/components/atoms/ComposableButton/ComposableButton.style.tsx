import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  wrapper: {
    borderStyle: 'solid',
    opacity: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent',
  },
  pressed: {
    opacity: 0.8,
  },
});
export default style;
