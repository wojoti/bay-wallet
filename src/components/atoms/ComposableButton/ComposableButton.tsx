import {PropsWithChildren} from 'react';
import {Pressable, StyleProp, ViewStyle} from 'react-native';
import style from './ComposableButton.style';

export type ComposableButtonProps = PropsWithChildren<{
  onPress: () => void;
  customStyle?: StyleProp<ViewStyle>;
  testId: string;
  disabled?: boolean;
}>;

const ComposableButton = ({
  onPress,
  customStyle,
  disabled = false,
  testId,
  children,
}: ComposableButtonProps) => {
  return (
    <Pressable
      accessibilityRole="button"
      testID={testId}
      style={({pressed}) => [
        style.wrapper,
        pressed && style.pressed,
        customStyle,
      ]}
      onPress={onPress}
      disabled={disabled}>
      {children}
    </Pressable>
  );
};

export default ComposableButton;
