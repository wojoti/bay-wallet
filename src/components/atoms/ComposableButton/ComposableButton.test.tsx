import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import {Text} from 'react-native';
import TestRenderer from 'react-test-renderer';
import ComposableButton, {ComposableButtonProps} from './ComposableButton';

const handlePress = jest.fn();
const props: ComposableButtonProps = {
  onPress: handlePress,
  testId: 'test-composablebutton-id',
  children: <Text>test composablebutton</Text>,
};

const modifiedProps: ComposableButtonProps = {
  ...props,
  disabled: true,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<ComposableButton {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render matching button with text', () => {
  render(<ComposableButton {...props} />);
  const buttonElement = screen.getByTestId('test-composablebutton-id');
  expect(buttonElement).toBeOnTheScreen();
  const buttonTextElement = within(buttonElement).getByText(
    'test composablebutton',
  );
  expect(buttonTextElement).toBeOnTheScreen();
});

test("shouldn't be able to handle onPress when disabled", () => {
  render(<ComposableButton {...modifiedProps} onPress={handlePress} />);
  const buttonElement = screen.getByTestId('test-composablebutton-id');
  fireEvent.press(buttonElement);
  expect(buttonElement).toBeDisabled();
  expect(handlePress).toHaveBeenCalledTimes(0);
});

test('should handle onPress event', () => {
  render(<ComposableButton {...props} />);
  const buttonElement = screen.getByTestId('test-composablebutton-id');
  fireEvent.press(buttonElement);
  expect(buttonElement).toBeEnabled();
  expect(handlePress).toHaveBeenCalledTimes(1);
});
