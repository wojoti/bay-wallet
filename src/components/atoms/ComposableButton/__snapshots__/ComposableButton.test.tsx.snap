// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderStyle": "solid",
        "justifyContent": "center",
        "opacity": 1,
      },
      false,
      undefined,
    ]
  }
  testID="test-composablebutton-id"
>
  <Text>
    test composablebutton
  </Text>
</View>
`;
